third_party/move/move-binary-format/src/file_format.rs

```diff
@@ -1340,501 +1341,1256 @@ pub struct CodeUnit {
-    /// Abort execution if such an object does not exist.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., address_value -> ..., value```
+
+    #[group = "global"]
+    #[static_operands = "[struct_def_idx]"]
+    #[description = r#"
+        Move the value of the specified type under the address in the global storage onto the top of the stack. 
+
+        Abort execution if such an value does not exist.    
+    "#]
+    #[semantics = r#"
+        stack >> addr
+
+        if global_state[addr] contains struct_type
+            stack << global_state[addr][struct_type]
+            global_state[addr][struct_type] = null
+        else
+            error
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> ty 
+        assert ty == signer 
+        assert struct_ty has key 
+        ty_stack << struct_ty
+    "#]
     MoveFrom(StructDefinitionIndex),
+    #[group = "global"]
+    #[static_operands = "[struct_inst_idx]"]
+    #[description = "Generic version of `MoveFrom`"]
+    #[semantics = "See `MoveFrom`."]
+    #[paranoid_post = "See `MoveFrom`."]
+    #[gas_type_creation_tier_0 = "resource_ty"]
     MoveFromGeneric(StructDefInstantiationIndex),
-    /// Move the instance at the top of the stack to the address of the `Signer` on the stack below
-    /// it
-    /// Abort execution if an object of type StructDefinitionIndex already exists in address.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., signer_value, value -> ...```
+
+    #[group = "global"]
+    #[static_operands = "[struct_def_idx]"]
+    #[description = r#"
+        Move the value at the top of the stack into the global storage,
+        under the address of the `signer` on the stack below it. 
+
+        Abort execution if an object of the same type already exists under that address.
+    "#]
+    #[semantics = r#"
+        stack >> struct_val
+        stack >> &signer
+        
+        if global_state[signer.addr] contains struct_type
+            error
+        else
+            global_state[signer.addr][struct_type] = struct_val
+    "#]
+    #[paranoid_post = r#"
+        ty_stack >> ty1 
+        assert ty1 == signer 
+        ty_stack >> ty2 
+        assert ty == struct_ty 
+        assert struct_ty has key
+    "#]
     MoveTo(StructDefinitionIndex),
+    #[group = "global"]
+    #[static_operands = "[struct_inst_idx]"]
+    #[description = "Generic version of `MoveTo`"]
+    #[semantics = "See `MoveTo`."]
+    #[paranoid_post = "See `MoveTo`."]
+    #[gas_type_creation_tier_0 = "resource_ty"]
     MoveToGeneric(StructDefInstantiationIndex),
-    /// Shift the (second top value) left (top value) bits and pushes the result on the stack.
-    ///
-    /// Stack transition:
-    ///
-    /// ```..., u64_value(1), u64_value(2) -> ..., u64_value```
+
+    #[group = "bitwise"]
+    #[description = r#"
+        Shift the (second top value) right (top value) bits and pushes the result on the stack.
```

Comment:
Are there any restrictions on the shifted value being `u64` (document it eitherway)?